### DISCUSSION TITLE
SSH not enabled after build

### DISCUSSION THREAD

**Discussion by arkadicolson** [vmware/packer-examples-for-vsphere#780] [Help]
Hi

After building my ubuntu22 image with this great tool I'm ubable to login through ssh over the network. After investigation I can see that the ssh service is reporting "no hostkeys available". So I ran sshkey-gen and tsrted the service.
What is the reason for this? Did I configure something wrong in the config?

Is there als also way to change the hostname and the vm name? Now hostname is localhost and vmname = linux-ubuntu-22.04-lts-develop

Thx!

**Comment by tenthirtyam** [comment]
Try against the last v23.10 release or main branch and let me know. develop branch is subject to issues pre-release.

**Reply by tenthirtyam** [reply]
Likely a result of the last Ansible refactoring update in the development branch.

**Reply by tenthirtyam** [reply]
Moved to https://github.com/vmware-samples/packer-examples-for-vsphere/issues/781

**Reply by arkadicolson** [reply]
I tried the main branch with unfortunately the same result. No ssh after boot. The problem seems to be related to the rc.local file. First line is a comment block from ansible which should not be there. Removing the first line, reboot and everything is ok.

# BEGIN ANSIBLE MANAGED BLOCK
#!/bin/bash
test -f /etc/ssh/ssh_host_dsa_key || dpkg-reconfigure openssh-server
exit 0
# END ANSIBLE MANAGED BLOCK

**Reply by tenthirtyam** [reply]
Try again with the latest in the develop branch.

**Comment by arkadicolson** [comment]
It's still not working. I realize now that the formatting is wrong in my previous message:
```shell
# BEGIN ANSIBLE MANAGED BLOCK
#!/bin/bash
test -f /etc/ssh/ssh_host_dsa_key || dpkg-reconfigure openssh-server
exit 0
# END ANSIBLE MANAGED BLOCK
``` 
should be

```shell
#!/bin/bash
test -f /etc/ssh/ssh_host_dsa_key || dpkg-reconfigure openssh-server
exit 0
```
As you can see both comment blocks from ansible shouln't be there.

**Comment by arkadicolson** [comment]
This worked for me:

```hcl
# Tasks for setting SSH keys to regenerate.
- name: "Setting SSH keys to regenerate."
  lineinfile:
    dest: /etc/rc.local
    create: true
    line: "#!/bin/bash\ntest -f /etc/ssh/ssh_host_dsa_key || dpkg-reconfigure openssh-server\nexit 0\n"
    mode: '0755'

```

**Reply by tenthirtyam** [reply]
Merged in #782
